### PR TITLE
feature(typescript): add TypeScript type definitions

### PR DESCRIPTION
**Pull request by igoro00**
add TypeScript definitions to library's API
for better integration in TS projects and better DX(autocompletion
and auto-importing)

### CODE REVIEW

**Comment by igoro00** [comment]
sorry that there are no tests for TS typings but I don't know how to implement them correctly

File: lib/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    string: string,
+    special: string,
+    bracket: string,
+    clear: string
+  }
+}
+
+declare module 'sql-highlight' {
+  export function highlight(sqlString: string, options?:highlightOptions): string;
+}
```
Comment (scriptcoded) [inline lib/index.d.ts:17]: Formatting according to my other TypeScript projects as well as capitalizing `highlightOptions`.

```suggestion
export interface HighlightOptions {
  html?: boolean;
  classPrefix?: string;
  colors?: {
    keyword: string;
    function: string;
    number: string;
    string: string;
    special: string;
    bracket: string;
    clear: string;
  };
}

declare module 'sql-highlight' {
  export function highlight(sqlString: string, options?: HighlightOptions): string
}

```

**Review by scriptcoded — CHANGES_REQUESTED** [review]
Hey, thanks for the PR!

Sorry, didn't notice you raised it as I got a bunch of notifications from https://github.com/pomahtuk/sequilize-highlight.

I'll have a closer look at the typings as well as tests tomorrow if I get the time. I took a quick peek and for now I just have a few formatting wishes :)

Another thing I'll just put out there. Should the options interface be inside the module declaration, and should there be a namespace? I usually don't write a lot of typings, so I'm not sure about the best approach. I want to be a bit careful here as modifying changes after the fact most likely would mean a breaking change, or at least potentially breaking users build pipelines.

**Comment by igoro00** [comment]
That interface should definitely be inside the module, my bad.
I don't know about the namespace though. I don't have much experience with writing declaration files either(I usually just write the code in TypeScript and the compiler generates definitions for me) but I looked at .d.ts files of many libraries that I currently use in my project(typegraphql+apollo+fastify and their dependencies) and >90% of them either top-level export everything or `
declare module 'name-of-the-module' {
  //all types exported here
}
`.

Also DefinitelyTyped in their [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#how-can-i-contribute) says that you should just `declare module`. They don't mention namespaces at all.

**Comment by igoro00** [comment]
>  I want to be a bit careful here as modifying types after the fact most likely would mean a breaking change, or at least potentially breaking users build pipelines.

I think as long type declarations match the code and we test if it builds correctly with a test project we should be perfectly fine. For example I tested right now(manually ofc) if moving the HighlightOptions interface to the module would break my project(I'm already using that .d.ts in my code by just pasting the file in node_modules) and typescript just didn't care. It built just fine and intellisence didn't change.

**Review by scriptcoded — CHANGES_REQUESTED** [review]
@igoro00 In that case I think these changes should be pretty much sufficient. If you can just add `"types": "./lib/index.d.ts"` to `package.json` (right after `"main": "./lib/index.js"`) so that NPM others can pick it up I'm happy to merge your changes.

As for testing the types I might add linting in the future, and perhaps also check that the types are updated if the exposed API is changed, if possible. I'll leave that be for now though.

Thanks for contributing!

**Pull request merged**